packages/transformer/src/test/standalone/IModelTransformerHub.test.ts

```diff
@@ -3255,6 +3255,188 @@ describe("IModelTransformerHub", () => {
+            jsonProperties: JSON.stringify({
+              pendingReverseSyncChangesetIndices: [],
+              pendingSyncChangesetIndices: [],
+              reverseSyncVersion: ";0", // not synced yet
+            }),
+          } as ExternalSourceAspectProps);
+          targetScopeProvenanceProps = targetScopeProvenance;
+        },
+      },
+      {
+        branch: {
+          manualUpdate(branch) {
+            // Check it fails without jsonprops
+            branch.elements.updateAspect({
+              ...targetScopeProvenanceProps!,
+              jsonProperties: undefined,
+            });
+          },
+        },
+      },
+      {
+        master: {
+          // Reverse sync and reverse sync looks for a 'reverseSyncVersion' inside of jsonProperties which will be missing so expectthrow.
+          sync: ["branch", { expectThrow: true }],
+        },
+      },
+      {
+        branch: {
+          sync: ["master", { expectThrow: false }],
+        },
+      },
+      {
+        branch: {
+          manualUpdate(branch) {
+            // Check it fails without version now.
+            branch.elements.updateAspect({
+              ...targetScopeProvenanceProps!,
+              version: undefined,
+            } as ExternalSourceAspectProps);
+          },
+        },
+      },
+      {
+        branch: {
+          // Forward sync and forward sync looks for a prop 'version' on the ESA which will be missing so expect to throw.
+          sync: ["master", { expectThrow: true }],
+        },
+      },
+      {
+        master: {
+          sync: ["branch", { expectThrow: false }],
+        },
+      },
+      {
+        branch: {
+          manualUpdate(branch) {
+            // Remove both and make sure it passes with both removed + setIgnoreNoBranchRelationshipData
+            branch.elements.updateAspect({
+              ...targetScopeProvenanceProps!,
+              jsonProperties: undefined,
+              version: undefined,
+            } as ExternalSourceAspectProps);
+          },
+        },
+      },
+      {
+        branch: {
+          sync: [
+            "master",
+            {
+              expectThrow: false,
+              initTransformer: setIgnoreNoBranchRelationshipData,
+            },
+          ],
+        },
+      },
+      {
+        master: {
+          sync: [
+            "branch",
+            {
+              expectThrow: false,
+              initTransformer: setIgnoreNoBranchRelationshipData,
+            },
+          ],
+        },
+      },
+      // eslint-disable-next-line @typescript-eslint/no-shadow
+      {
+        assert({ master, branch }) {
+          expect(master.db.changeset.index).to.equal(5);
+          expect(branch.db.changeset.index).to.equal(8);
+          expect(count(master.db, ExternalSourceAspect.classFullName)).to.equal(
+            0
+          );
+          // added because the root was modified
```

Comment:
what was added? How many were added? can you redefine `11` in terms of the amount there were before `+` something?

Reply:
I also copied this from another test, but I just added a comment explaining 11 ESAs. Let me know your thoughts

Reply:
why not just:
```ts
expect(count(branch.db, "bis.ExternalSourceAspect")).to.equal(
       count(branch.db, "bis.Element"));
```
I think that better indicates it. In general comments like this can become stale easily, it's best to "comment assumptions with code" in tests

Reply:
I agree that seems better, but I also think theres value in me fully explaining the contents of the database at that point in time. Maybe this is the wrong place to do it, but somewhere is better than nowhere imo. I think its one small step towards better understanding of the bis stuff in general

If the number were to change from 11, I'd probably want to know for my own enrichment because maybe that means theres another element? 

I did change the comment around a bit just now, but not significantly imo.

Reply:
I still think a comment is a future source of staleness.
What do you think about converting your comment into:
```ts
const externalAspectCounts = (db) => db.query(`
  SELECT e.ECInstanceId, COUNT(*) FROM bis.ExternalSourceAspect esa
  JOIN bis.Element e ON e.ECInstanceId=esa.Element.Id
  GROUP BY e.ECInstanceId
`);
  

expect(count(db, "bis.ExternalSourceAspect")).to.be(11);
expect(count(db, "bis.Element)).to.be(10);

for (const [id, aspectCount] of externalAspectCounts(db)) {
  if (id === transformer.targetScopeElement)
    expect(aspectCount).to.equal(2);
  else
    expect(aspectCount).to.equal(1);
}
```

The full db contents really belong not here but elsewhere, earlier in the `it` scope, I assume right after initialization of the db

Reply:
I see now you wanted more "at that point in time".
really imo the need for this is due to poor diffing support elsewhere

Reply:
The comment being a source of staleness is a good thing imo, we should understand when our database suddenly has 12 elements instead of 11. I removed the comment 

Yeah agree about the diffing support, what I've been doing locally is modifying the count helper function to push rows to an array that I can look at in the debugger. Not sure if you're opposed to me committing that as I think it would be slower. ( I wouldn't add it in this PR just some other time)

Reply:
feel free to commit it under an environment variable based branch